docs: add docs for waterfall chart (#74163)
metabase/metabase: docs/questions/visualizations/waterfall-chart.md

# Waterfall charts

Waterfall charts are a type of bar chart for visualizing data with both positive and negative values. Each bar shows an increase or decrease, and a final bar on the right shows the running total.

![Waterfall chart](../images/waterfall-chart.png)

## When to use a waterfall chart

Use a waterfall chart to show how a starting value changes through a series of increases and decreases to arrive at a final total. 

Waterfall charts are useful for financial breakdowns, inventory changes, or anywhere you need to show how a value changes step by step.

## Data shape for a waterfall chart

> To create a waterfall chart, create a question that returns a single metric grouped by a single dimension, such as time or category. The metric should include both positive and negative values.

Reviewer: > The metric should include both positive and negative values.

This isn't true - it doesn't _need_ to include positive and negative values. It can just be positive values:
<img width="2668" height="1726" alt="image" src="https://github.com/user-attachments/assets/f93de963-fbbf-44c1-95da-197555cf43d4" />

Here's the data shape for the example chart above:

> | Month      | Change in sales |

Reviewer: Great call on calling the column "change in sales"! I think we should also explicitly emphasize in text that you probable (although not _necessarily_) want to have the metric column contain the _change_ in some other metric, not that other metric itself.

The reason why this is worth mentioning is that usually data in real world does _not_ come in this form. You might have revenue for April, May, June in your data, but you usually do not have the _change_ from April to May, May to June etc in your data, and you'd need to do some prep work to make that happen. 

We should explain that, and ideally give people the explanation of how to do that prep work (e.g. by using `Offset` custom expression to get the change between previous and next row). See docs for Funnel chart for another chart that you need to do the pre-work for.

| ---------- | --------------- |
| January    |  12,500         |
| February   | -3,200          |
| March      |  5,800          |
| April      | -1,600          |
| May        |  7,100          |
| June       |  4,200          |

Green bars represent months where sales increased, and red bars represent months where sales decreased. The dark bar at the end shows the net change across all six months.

Here is another example grouped by category instead of time:

| Category    | Net revenue |
| ----------- | ----------- |
| Gizmo       | 18,400      |
| Doohickey   | -2,100      |
| Widget      | 9,700       |
| Gadget      | -4,500      |

## Waterfall chart settings

To open chart settings, click the **Gear** icon in the bottom left of the visualization.

### Display

> In chart settings, click the **Display** tab to edit the chart display.

Reviewer: I'm not sure what "Chart display" means

> To add a goal line to your waterfall chart, enable the **Goal line** toggle. Use the **Goal value** and **Goal label** fields to set the goal value and label.

Reviewer: Please add here and in the Limitations session that on the waterfall chart, you _can't_ set an alert for the goal line, and link the alert docs

# Waterfall charts

Waterfall charts are a type of bar chart for visualizing data with both positive and negative values. Each bar shows an increase or decrease, and a final bar on the right shows the running total.

![Waterfall chart](../images/waterfall-chart.png)

## When to use a waterfall chart

Use a waterfall chart to show how a starting value changes through a series of increases and decreases to arrive at a final total. 

Waterfall charts are useful for financial breakdowns, inventory changes, or anywhere you need to show how a value changes step by step.

> If your data only contains positive values, or if you do not need to track a running total, consider a [bar chart](./line-bar-and-area-charts.md) instead.

Reviewer 2: ```suggestion
```
We mention this in limitations.

## Data shape for a waterfall chart

> To create a waterfall chart, use a query that returns a single metric grouped by a single dimension, such as time or category. The metric should include both positive and negative values.

Reviewer 2: ```suggestion
To create a waterfall chart, create a question that returns a single metric grouped by a single dimension, such as time or category. The metric should include both positive and negative values.
```

> Here is the data shape for the example chart on this page:

Reviewer 2: ```suggestion
Here's the data shape for the example chart above:
```

| Category    | Net revenue |
| ----------- | ----------- |
| Gizmo       | 18,400      |
| Doohickey   | -2,100      |
| Widget      | 9,700       |
| Gadget      | -4,500      |

## Waterfall chart settings

To open chart settings, click the **Gear** icon in the bottom left of the visualization.

### Display

In chart settings, click the **Display** tab to edit the chart display.

To add a goal line to your waterfall chart, enable the **Goal line** toggle. Use the **Goal value** and **Goal label** fields to set the goal value and label.

Set the colors for each bar type:

- **Increase color:** The color for bars that represent positive values (defaults to green)
- **Decrease color:** The color for bars that represent negative values (defaults to red)
- **Total color:** The color for the final total bar (defaults to black)

To display the value for the final total bar, enable the **Show total** toggle.

To display values for all bars, enable the **Show values on data points** toggle.

### Axes

In chart settings, click the **Axes** tab to edit the chart axes.

#### X-axis

To display the x-axis label, enable the **Show label** toggle.

> Use the following options for the x-axis:

Reviewer 2: ```suggestion
```
Could just tack on an "Options include:" to end the previous paragraph, but not necessary.

Author: Not all the options on the list are related to the axis label, so maybe it makes sense to keep this line break? I could also move the "Show label" toggle to the list of "options" so everything's in one place

- **Label:** Name the x-axis label.
- **Show lines and tick marks:** Display the x-axis line and tick marks.

#### Y-axis

To display the y-axis label, enable the **Show label** toggle.

> Use the following options for the y-axis:

Reviewer 2: ```suggestion
```

Author: same as above

- **Label:** Name the y-axis label.
- **Auto y-axis range:** When enabled, Metabase sets the y-axis range automatically based on your data. When disabled, use the **Min** and **Max** fields to set custom values.
- **Scale:** Choose how values are spaced along the y-axis. Choose between **Linear**, **Power**, and **Log**.
- **Show lines and tick marks:** Display the y-axis line and tick marks.
- **Number of tick marks:** Set how many tick marks appear on the y-axis. Defaults to **auto**.

## Limitations and alternatives

> - If your data only contains positive values, or if you do not need to track a running total, consider a [bar chart](./line-bar-and-area-charts.md) instead.

Reviewer 2: ```suggestion
- If your data only contains positive values, or if you don't need to track a running total, consider a [bar chart](./line-bar-and-area-charts.md) instead.
```

- If you want to show progress toward a single value, consider a [progress bar](./progress-bar.md) or [gauge chart](./gauge.md) instead.

> - Waterfall charts only support one metric grouped by one dimension. You cannot add breakouts or display multiple series.

Reviewer 2: ```suggestion
- Waterfall charts only support one metric grouped by one dimension. You can't add breakouts or display multiple series.
```

> - Waterfall charts only support one metric grouped by one dimension. You cannot add breakouts or display multiple series.

Reviewer 2: ```suggestion
- Waterfall charts only support a single metric grouped by a single dimension. You can't add breakouts or display multiple series.
```
